### PR TITLE
[EMB-444][EMB-446][EMB-428][EMB-433][EMB-464] Reg overview (mostly)

### PR DESCRIPTION
**Pull request by aaxelb**
## Ticket

https://openscience.atlassian.net/browse/EMB-

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/registries/addon/components/link/component.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+@tagName('')
+@layout(template)
+export default class Link extends Component {
+    @service router!: RouterService;
+
+    route!: string;
+    models?: any[];
+    queryParams?: Record<string, any>;
+
+    tag: string = defaultTo(this.tag, 'a');
```
Comment (aaxelb) [inline lib/registries/addon/components/link/component.ts]: unused?

File: lib/registries/addon/components/link/component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+type AnchorRel = 'noreferrer' | 'noopener';
+type AnchorTarget = '_self' | '_blank' | '_parent' | '_top';
+
+@tagName('')
+@layout(template)
+export default class Link extends Component {
+    @service router!: RouterService;
+
+    route!: string;
```
Comment (aaxelb) [inline lib/registries/addon/components/link/component.ts]: should be optional

File: lib/registries/addon/components/link/component.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @computed('router.currentURL')
+    get isActive() {
+        return Boolean(this.route) && this.router.isActive(this.route, ...(this.models || []), {
+            queryParams: this.queryParams || {},
+        });
+    }
+
+    didReceiveAttrs() {
```
Comment (aaxelb) [inline lib/registries/addon/components/link/component.ts]: assert `route` or `href` provided (if `href` passed as an attribute is accessible)

File: lib/registries/addon/components/link/template.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<a
+    onclick={{action this._onclick}}
+    href={{this.href}}
+    target={{this.target}}
+    data-active={{this.isActive}}
```
Comment (aaxelb) [inline lib/registries/addon/components/link/template.hbs]: remove?

File: lib/registries/addon/components/registration-form-view/x-text/template.hbs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<p>{{this.part.value}}</p>
+
```
Comment (aaxelb) [inline lib/registries/addon/components/registration-form-view/x-text/template.hbs]: blank line

File: lib/registries/addon/overview/contributors/template.hbs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    relationshipName='contributors'
+    as |list|
+}}
+    {{#list.item as |contributor|}}
+        {{#if contributor}}
+            <div style='display: flex'>
+                {{! TODO: user card? }}
+                <img data-test-gravatar
+                     alt={{t 'registries.overview.contributors.gravatar_alt' name=contributor.users.fullName}}
+                     style='height: 30px; width: 30px'
```
Comment (aaxelb) [inline lib/registries/addon/overview/contributors/template.hbs]: get rid of inline style

File: lib/registries/addon/overview/controller.ts [hunk 7]
```diff
@@ -2,15 +2,36 @@ import { action, computed } from '@ember-decorators/object';
...
 
+import Registration from 'ember-osf-web/models/registration';
+
 export default class Overview extends Controller {
     @service media!: Media;
     @not('media.isDesktop') showMobileNav!: boolean;
 
+    // Model could be a Registration from a transition or wrapped in a task
+    // from the route
+    model!: Registration | { taskInstance: TaskInstance<Registration> };
```
Comment (aaxelb) [inline lib/registries/addon/overview/controller.ts]: `GuidRouteModel<Registration>`

File: lib/registries/addon/overview/forks/route.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@ import Route from '@ember/routing/route';
 
 export default class ForksPage extends Route {
     beforeModel() {
-        const { registrationId: guid } = (this.modelFor('overview') as { registrationId: string });
-        this.transitionToExternal('guid-registration.forks', guid);
+        const model = this.modelFor('overview') as any;
```
Comment (aaxelb) [inline lib/registries/addon/overview/forks/route.ts]: `GuidRouteModel<Registration>`

File: lib/registries/addon/overview/index/controller.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { computed } from '@ember-decorators/object';
+import Controller from '@ember/controller';
+import { TaskInstance } from 'ember-concurrency';
+import Registration from 'ember-osf-web/models/registration';
+
+export default class Overview extends Controller {
+    // Model could be a Registration from a transition or wrapped in a task
+    // from the route
+    model!: Registration | { taskInstance: TaskInstance<Registration> };
```
Comment (aaxelb) [inline lib/registries/addon/overview/index/controller.ts]: `GuidRouteModel<Registration>`

File: tests/engines/registries/acceptance/overview/index-test.ts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+module('Registries | Acceptance | overview.index', hooks => {
+    setupEngineApplicationTest(hooks, 'registries');
+    setupMirage(hooks);
+
+    hooks.beforeEach(function(this: TestContext) {
+        server.create('root', { currentUser: null });
+    });
+
+    test('it renders', async function(this: TestContext, assert: Assert) {
+        const registration = server.create('registration', 'withRegisteredMeta');
```
Comment (aaxelb) [inline tests/engines/registries/acceptance/overview/index-test.ts]: this doesn't seem to work? check percy

**Comment by brianjgeiger** [comment]
Sorry about that. I thought I had changed `setupEngineApplicationTest` to call `setupOSFApplicationTest`, but I just checked develop and it wasn't set.

**Pull request merged**